kalebswartz7/excercise_01/parse.py

```diff
@@ -0,0 +1,136 @@
+# Parse the data from the dat file -> Points vs Horizontal Position = how many rows per column in the 2D array 
+#                                  -> Points vs Vertical Position = how many columns in the 2D array 
+
+#       Create a dictionary and populate it with other dictionaries based on the data in the dat file 
+#       Use this data to add to the graph
+
+import matplotlib as mpl
+import numpy as np
+import matplotlib.pyplot as plt
+
+#Just default values, these are changed based on the dat file
+horizontal = 1 
+vertical = 1
+
+#initialize the dictionary for storing parameters given in the dat file 
+initialData = {}
+initialData['photon'] = {}
+initialData['horizontal'] = {}
+initialData['vertical'] = {}
+matrix = [[]] 
+
+""" 
+Get data from the dat file with the parameters for the scan. Put this data into an organized Dictionary
+
+"""
+def getCommentedData():
+    with open('res_int_pr_se.dat') as file:
+        lineNum = 1
+        for line in file:
+            if "#" not in line:
+                return
+            else:
+                if (lineNum is 1):
+                    populateDictionary(line, '[', ']', 'intensity')
+                elif ('Initial Photon' in line):
+                    populateDictionary(line, '#', ' ', 'photon', 'Initial_Energy')
+                elif ('Final Photon' in line):
+                    populateDictionary(line, '#', ' ', 'photon', 'Final_Energy')
+                elif ('Initial Horizontal' in line):
+                    populateDictionary(line, '#', ' ', 'horizontal', 'Initial_Position')
+                elif ('Final Horizontal' in line):
+                    populateDictionary(line, '#', ' ', 'horizontal', 'Final_Position')
+                elif ('vs Horizontal' in line):
+                    populateDictionary(line, '#', ' ', 'horizontal', 'Points', 'float','horizontal')
+                elif ('Initial Vertical' in line):
+                    populateDictionary(line, '#', ' ', 'vertical', 'Initial_Position')
+                elif ('Final Vertical' in line):
+                    populateDictionary(line, '#', ' ', 'vertical', 'Final_Position')
+                elif ('vs Vertical' in line):
+                    populateDictionary(line, '#', ' ', 'vertical', 'Points', 'float','vertical')
+
+
```

Comment:
It's not recommended to have 2 blank lines within a function/class. They are normally used to separate different classes/functions.

Reply:
This has also been fixed.